decidim-core/app/controllers/decidim/participatory_processes_controller.rb

```diff
@@ -6,15 +6,14 @@ module Decidim
   # A controller that holds the logic to show ParticipatoryProcesses in a
   # public layout.
   class ParticipatoryProcessesController < ApplicationController
-    include NeedsParticipatoryProcess
-
     layout "layouts/decidim/participatory_process", only: [:show]
 
-    skip_after_action :verify_participatory_process, only: [:index]
+    before_action -> { extend(NeedsParticipatoryProcess) }, only: [:show]
```

Comment:
Thanks! Still, the approach looks a little weird though. Why can't this be done with a `skip_after_action`, or maybe providing a custom class method that provides similar results?